compiler/grainlsp/utils.re

```diff
@@ -0,0 +1,816 @@
+open Grain;
+open Compile;
+open Grain_parsing;
+open Grain_utils;
+open Grain_typed;
+open Grain_diagnostics;
+
+//  CompletionItemKind
+let item_kind_completion_text = 1;
+let item_kind_completion_function = 3;
+let item_kind_completion_constructor = 4;
+let item_kind_completion_variable = 6;
+let item_kind_completion_struct = 22;
+
+let markdown_grain = code => "```grain\n" ++ code ++ "\n```";
+
+type node_t =
+  | Expression(Typedtree.expression)
+  | Pattern(Typedtree.pattern)
+  | NotInRange
+  | Error(string);
+
+type source_text =
+  | Nothing
+  | Lident(string);
+
+let rec print_path = (ident: Path.t) => {
+  switch (ident) {
+  | PIdent(name) => name.name
+  | PExternal(externalIdent, second, _) =>
+    print_path(externalIdent) ++ "." ++ second
+  };
+};
+
+let comma_join = (~sep, ~print_fn, vals) => {
```

Comment:
This is basically String.concat?

Reply:
With a map of the args, so simplified to use that as the implementation